neuro_san/service/main_loop/server_main_loop.py

```diff
@@ -158,34 +160,50 @@ def main_loop(self):
         metadata_set = metadata_set | set(self.usage_logger_metadata.split())
         metadata_str: str = " ".join(sorted(metadata_set))
 
-        self.server = GrpcAgentServer(self.port,
-                                      server_loop_callbacks=self,
-                                      network_storage_dict=self.network_storage_dict,
-                                      agent_networks=self.agent_networks,
-                                      server_name=self.server_name,
-                                      server_name_for_logs=self.server_name_for_logs,
-                                      max_concurrent_requests=self.max_concurrent_requests,
-                                      request_limit=self.request_limit,
-                                      forwarded_request_metadata=metadata_str)
+        self.grpc_server = GrpcAgentServer(self.grpc_port,
+                                           server_loop_callbacks=self,
+                                           network_storage_dict=self.network_storage_dict,
+                                           server_status=self.server_status,
+                                           server_name=self.server_name,
+                                           server_name_for_logs=self.server_name_for_logs,
+                                           max_concurrent_requests=self.max_concurrent_requests,
+                                           request_limit=self.request_limit,
+                                           forwarded_request_metadata=metadata_str)
+        self.grpc_server.prepare_for_serving()
 
```

Comment:
Split gRPC server logic into create+prepare (register itself as a listener) and then actually serve.

Reply:
Suggestion: if the grpc_port value is 0, then don't even start the GrpcServer.
Same deal for http_port